core/src/main/java/org/jivesoftware/AccountCreationWizard.java

```diff
@@ -353,7 +360,23 @@ private XMPPConnection getConnection() throws SmackException, IOException, XMPPE
         {
             builder.setHost( localPreferences.getXmppHost() );
         }
-
+        
+        if (securityMode != ConnectionConfiguration.SecurityMode.disabled && !useOldSSL) {
+            builder.setPort(5222);
```

Comment:
Changed. I digged into that a bit more and while 5222 is recommended because other ports might be blocked by firewalls, there is no general prohibition against using other ports.